tofd/pars/R3BTofDCal2HitPar.cxx

```diff
@@ -297,1099 +352,2668 @@ void R3BTofDCal2HitPar::Exec(Option_t* option)
-                                                             // parq[3] * pow(posToT, 3));
-                        qb = qb *
-                             fTofdQ; // theory says: dE ~ Z^2 but we see quenching -> just use linear and fit the rest
-                        // std::cout<< " hola30 " << qb << " , " << fTofdQ << std::endl;
-                    }
-                    else
-                    {
-                        // double exponential
-                        auto pos = ((bot_ns + par->GetOffset1()) - (top_ns + par->GetOffset2())) * par->GetVeff();
-                        auto q1 = bot_tot /
-                                  (parq[0] * (exp(-parq[1] * (pos + 100.)) + exp(-parq[2] * (pos + 100.))) + parq[3]);
-                        parq[0] = par->GetPar2a();
-                        parq[1] = par->GetPar2b();
-                        parq[2] = par->GetPar2c();
-                        parq[3] = par->GetPar2d();
-                        auto q2 = top_tot /
-                                  (parq[0] * (exp(-parq[1] * (pos + 100.)) + exp(-parq[2] * (pos + 100.))) + parq[3]);
-                        q1 = q1 *
-                             fTofdQ; // theory says: dE ~ Z^2 but we see quenching -> just use linear and fit the rest
-                        q2 = q2 * fTofdQ;
-                        qb = (q1 + q2) / 2.;
-                    }
-
-                    // Fill control histograms and Q vs Pos without multihits
-                    if (multihits[iPlane - 1][iBar - 1] < 2 && (qb > 0.))
-                    {
-                        // std::cout<< " h: " << posToT << " , " << qb << std::endl;
-                        fhQvsPos[iPlane - 1][iBar - 1]->Fill(posToT, qb);
-                    }
-                }
+                    if((sqrt(top_tot*bot_tot)>fToTMin &&sqrt(top_tot*bot_tot)<fToTMax) || ToTexeption){
+		    fhTsync[iPlane - 1]->Fill(iBar, THit);
+		    }
+		    // Tof with respect LOS detector
+		    auto tof = fTimeStitch->GetTime((top_ns + bot_ns) / 2. - fHeader->GetTStart(), "tamex", "vftx");
+		    if((sqrt(top_tot*bot_tot)>fToTMin &&sqrt(top_tot*bot_tot)<fToTMax) || ToTexeption){
+			    fh1_tofsync[iPlane - 1][iBar - 1]->Fill(tof);
+			    fhTofsync[iPlane-1]->Fill(iBar,tof);
+			    //cout<<"top_ns--> "<<top_ns<<endl;
+			    //	if(sqrt(top_tot*bot_tot)>fToTMin && sqrt(top_tot*bot_tot)<fToTMax){
+			    fh1_walk_t[iPlane-1][iBar-1]->Fill(top_tot,top_ns - fHeader->GetTStart());
+			    fh1_walk_b[iPlane-1][iBar-1]->Fill(bot_tot,bot_ns - fHeader->GetTStart());
+
+		    }
+		    //auto lost = fHeader->GetTStart();
+		    // if(tof>-153.){
+		    // cout<<"sqrtToT"<<sqrt(top_tot*bot_tot)<<endl; 
+		    //}
+		    //fhlos[iPlane-1]->Fill(iBar,lost);
+		    // if(iPlane==1 && iBar==11){
+		    //fhlost[1][11]->Fill(THit,lost);}
+		    //fhtott[iPlane-1][iBar-1]->Fill(THit,top_tot);
+		    }
+			
+			if(fTofdQ == /*0.0*/ 8. && fParameter == 10){
```

Comment:
8 was the Z of the calibration beam. It's an experiment specific variable. It could probably be changed to fTofdQ > 0 since fTofdQ is set in the macro.

There is no explanation of what fParameter=10 is.

Reply:
fParameter == 10 is a relic and can be removed.

fTofdQ is a flag if it is set to any value > 0 in the macro, it's basically a boolian statement at this point and should be treated as such.